File: dropwizard/src/main/java/com/yammer/dropwizard/jersey/params/IntParam.java
package com.yammer.dropwizard.jersey.params;

// TODO: 11/14/11 <coda> -- document IntParam
// TODO: 11/14/11 <coda> -- test IntParam

public class IntParam extends AbstractParam<Integer> {
    public IntParam(String input) {
        super(input);
    }

    @Override
    protected String errorMessage(String input, Exception e) {
        return '"' + input + "\" is not a number.";
    }

    @Override
    protected Integer parse(String input) {
        return Integer.valueOf(input);
    }
}

File: dropwizard-scala_2.9.1/src/main/scala/com/yammer/dropwizard/scala/params/IntParam.scala
package com.yammer.dropwizard.scala.params

object IntParam {
  def apply(value: Int): IntParam = IntParam(value.toString)
}

/**
 * Parses ints.
 */
case class IntParam(s: String) extends AbstractParam[Int](s) {
  protected def parse(input: String) = input.toInt

  override protected def renderError(input: String, e: Throwable) =
    "Invalid parameter: %s (Must be an integer value.)".format(input)
}
